python/src/trezorlib/cli/firmware.py

```diff
@@ -230,98 +268,54 @@ def version_str(version):
     return url, fingerprint
 
 
-def download_data_and_fingerprint_from_SL_servers(
-    version: str,
-    url: str,
-    model: str,
-    bootloader_version: list,
-    fingerprint: str,
-    beta: bool,
-    bitcoin_only: bool,
-) -> tuple:
-    """Gets the firmware data and its fingerprint, if that was not given"""
-    if not url:
-        if version:
-            version_list = [int(x) for x in version.split(".")]
-            if version_list[0] != bootloader_version[0]:
-                click.echo(
-                    "Warning: Trezor {} firmware version should be {}.X.Y (requested: {})".format(
-                        model, bootloader_version[0], version
-                    )
-                )
-        else:
-            version_list = None
-
-        url, fp = find_best_firmware_version(
-            bootloader_version=list(bootloader_version),
-            requested_version=version_list,
-            beta=beta,
-            bitcoin_only=bitcoin_only
-        )
-        # Replacing the fingerprint if it was not defined
-        if not fingerprint:
-            fingerprint = fp
-
+def download_firmware_data(url) -> bytes:
     try:
         click.echo("Downloading from {}".format(url))
         r = requests.get(url)
         r.raise_for_status()
-        firmware_data = r.content
+        return r.content
     except requests.exceptions.HTTPError as err:
         click.echo("Error downloading file: {}".format(err))
         sys.exit(3)
 
-    return firmware_data, fingerprint
-
 
-def check_the_validity_of_the_firmware(
+def validate_firmware(
     firmware_data: bytes,
     fingerprint: str = None,
     bootloader_onev2: bool = None,
     trezor_major_version: int = None,
 ) -> None:
-    """Validates the firmware through multiple tests:
-        - parsing it properly
-        - containing valid signatures and fingerprint (when chosen)
-        - being compatible with the device (when chosen)
+    """Validate the firmware through multiple tests.
+
+    - parsing it properly
+    - containing valid signatures and fingerprint (when chosen)
+    - being compatible with the device (when chosen)
     """
     try:
         version, fw = firmware.parse(firmware_data)
-        click.echo("Firmware is syntactically correct.")
     except Exception as e:
         click.echo(e)
         sys.exit(2)
 
-    recognize_and_print_firmware_version(
-        version=version,
-        fw=fw
-    )
-    validate_signatures_on_firmware(
-        version=version,
-        fw=fw
-    )
-    validate_fingerprint_on_firmware(
-        version=version,
-        fw=fw,
-        expected_fingerprint=fingerprint,
-    )
-    click.echo("Firmware is well signed and fingerprinted.")
+    common_kwargs = {'version': version, 'fw': fw}
```

Comment:
please instead just past `version, fw` as positional arguments in all of the calls below